pkg/cloudprovider/aws/launchtemplate.go

```diff
@@ -106,20 +106,14 @@ func (p *LaunchTemplateProvider) Get(ctx context.Context, constraints *v1alpha1.
 	}
 	// Construct launch templates
 	launchTemplates := map[string][]cloudprovider.InstanceType{}
+	caBundle, err := p.GetCABundle(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("getting ca bundle for user data, %w", err)
+	}
 	for amiID, instanceTypes := range amis {
-		// Get userData for Node
-		var userData string
-		if constraints.AMIFamily != nil {
-			if strings.EqualFold(*constraints.AMIFamily, v1alpha1.OperatingSystemBottleRocket) {
-				userData, err = p.getBottleRocketUserData(ctx, constraints, additionalLabels)
-			} else if strings.EqualFold(*constraints.AMIFamily, v1alpha1.OperatingSystemEKSOptimized) {
-				userData, err = p.getEKSOptimizedUserData(ctx, constraints, instanceTypes, additionalLabels)
-			} else {
-				logging.FromContext(ctx).Warnf("AMIFamily was set, but was not one of %s or %s. Setting to %s as the default.", v1alpha1.OperatingSystemEKSOptimized, v1alpha1.OperatingSystemBottleRocket, v1alpha1.OperatingSystemEKSOptimized)
-				userData, err = p.getEKSOptimizedUserData(ctx, constraints, instanceTypes, additionalLabels)
-			}
-		} else {
-			userData, err = p.getEKSOptimizedUserData(ctx, constraints, instanceTypes, additionalLabels)
+		userData := p.getEKSOptimizedUserData(ctx, constraints, instanceTypes, additionalLabels, caBundle)
```

Comment:
totally minor, but if you factored this out into a function, you could use the same return pattern used in the ami logic and keep this core Get() function as simple as possible.

Reply:
How's this?